packages/dart/npt_flutter/lib/features/profile/bloc/profile_bloc.dart

```diff
@@ -158,26 +162,20 @@ class ProfileBloc extends LoggingBloc<ProfileEvent, ProfileState> {
         emit(ProfileStarting(uuid, profile: profile, status: err));
       });
 
-      cancelSubs = () {
+      cancel = () {
         progressSub?.cancel();
         progressSub = null;
 
         errorSub?.cancel();
         errorSub = null;
+
+        if (sc is SocketConnector) sc.close();
       };
 
-      sc = await npt
-          .runInline()
-          // Todo - make this timeout configurable from settings
-          .timeout(
-        const Duration(seconds: 10),
-        onTimeout: () {
-          return TimedOutSocketConnector();
-        },
-      );
+      sc = await npt.runInline();
```

Comment:
If the timeout interrupts npt and the underlying socketConnector while the socket is bound, then the socket will stay bound and we have no surface to unbind it from here, hence I have removed the timeout.